File: Web/Access2Justice.Shared/Models/Navigation.cs
using Microsoft.AspNetCore.JsonPatch.Operations;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Swashbuckle.AspNetCore.Swagger;
using Swashbuckle.AspNetCore.SwaggerGen;
using System.Globalization;
using System.Text;
using System.ComponentModel;

namespace Access2Justice.Shared.Models
{    
    public class Navigation: NameLocation
    {
        [DefaultValue("")]
        [JsonProperty(PropertyName = "language")]
        public Language Language { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "locationNavContent")]
        public HeaderLocation LocationNav { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "privacyPromise")]
        public ButtonImage PrivacyPromise { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "helpAndFAQ")]
        public ButtonImage HelpAndFAQ { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "login")]
        public ButtonImage Login { get; set; }

        [JsonProperty(PropertyName = "logo")]
        public Logo Logo { get; set; }
        
        [DefaultValue("")]
        [JsonProperty(PropertyName = "home")]
        public ButtonStaticContent Home { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "guidedAssistant")]
        public ButtonStaticContent GuidedAssistant { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "topicAndResources")]
        public ButtonStaticContent TopicAndResources { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "about")]
        public ButtonStaticContent About { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "search")]
        public ButtonImage Search { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "organizationalUnit")]
        public string OrganizationalUnit { get; set; }
    }

    public class Language
    {
        [DefaultValue("")]
        [JsonProperty(PropertyName = "button")]
        public ButtonStaticContent Button { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "navigationImage")]
        public Image NavigationImage { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "dropDownImage")]
        public Image DropDownImage { get; set; }
    }

    public class HeaderLocation
    {
        [DefaultValue("")]
        [JsonProperty(PropertyName = "text")]
        public string Text { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "altText")]
        public string AltText { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "button")]
        public ButtonStaticContent Button { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "image")]
        public Image Image { get; set; }
    }

    public class ButtonImage
    {
        [DefaultValue("")]
        [JsonProperty(PropertyName = "button")]
        public ButtonStaticContent Button { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "image")]
        public Image Image { get; set; }
    }

    public class Logo
    {
        [DefaultValue("")]
        [JsonProperty(PropertyName = "firstLogo")]
        public string FirstLogo { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "secondLogo")]
        public string SecondLogo { get; set; }

        [DefaultValue("")]
        [JsonProperty(PropertyName = "link")]
        public string Link { get; set; }
    }
}

File: Web/Access2Justice.Web/ClientApp/src/app/shared/navigation/navigation.ts
export interface Navigation {
  name: string;
  location: Array<Location>;
  language: Language;
  locationNavContent: LocationNavContent;
  privacyPromise: PrivacyPromise;
  helpAndFAQ: HelpAndFAQ;
  login: Login;
  logo: Logo;
  home: Home;
  guidedAssistant: GuidedAssistant;
  topicAndResources: TopicAndResources;
  about: About;
  search: Search;
}

export interface LocationNavContent {
  text: string;
  altText: string;
  button: Button;
  image: Image
}

export interface Button {
    buttonText: string;
    buttonAltText: string;
    buttonLink: string;
}

export interface Image {
  source: string;
  altText: string;
}

export interface Language {
  button: Button;
  navigationImage: Image;
  dropDownImage: Image
}

export interface Location {
  state: string;
  city: string;
  county: string;
  zipCode: string;
  locality: string;
  address: string;
}

export interface Logo {
  firstLogo: string;
  secondLogo: string;
  link: string;
}

export interface PrivacyPromise {
  button: Button;
  image: Image;
}

export interface HelpAndFAQ {
  button: Button;
  image: Image;
}

export interface Login {
  button: Button;
  image: Image;
}

export interface Home {
  button: Button;
}

export interface GuidedAssistant {
  button: Button;
}

export interface TopicAndResources {
  button: Button;
}

export interface About {
  button: Button;
}

export interface Search {
  button: Button;
  image: Image;
}
